src/python/bot/fuzzers/afl/engine.py

```diff
@@ -50,25 +73,55 @@ def fuzz(self, target_path, options, reproducers_dir, max_time):
     Args:
       target_path: Path to the target.
       options: The FuzzOptions object returned by prepare().
-      reproducers_dir: The directory to put reproducers in when crashes
-          are found.
+      reproducers_dir: The directory to put reproducers in when crashes are
+        found.
       max_time: Maximum allowed time for the fuzzing to run.
 
    Returns:
       A FuzzResult object.
     """
-    config = launcher.AFLConfig.from_target_path(target_path)
+    environment.set_value('AFL_ENGINE_TIMEOUT', max_time)
+
+    config = launcher.AflConfig.from_target_path(target_path)
     config.additional_afl_arguments = options.arguments
-    input_path = None  # Not required for fuzzing.
+    testcase_file_path = os.path.join(reproducers_dir, 'testcase')
+
     runner = launcher.prepare_runner(
         target_path,
         config,
-        input_path,
+        testcase_file_path,
         options.corpus_dir,
         strategy_dict=options.strategies)
 
-    # TODO(mbarbella): Use the runner to start a fuzzing session.
-    raise NotImplementedError
+    fuzz_result = runner.fuzz()
+
+    command = _get_command_from_fuzz_result(fuzz_result, runner)
+    time_executed = fuzz_result.time_executed
+    logs = fuzz_result.output.splitlines() + runner.fuzzer_stderr.splitlines()
+
+    # Bail out if AFL returns a nonzero status code.
+    if fuzz_result.return_code:
+      return engine.FuzzResult(logs, command, [], {}, time_executed)
```

Comment:
nit: log_error to match libFuzzer: https://github.com/google/clusterfuzz/blob/0b5c023eca9e3aac41faba17da8f341c0ca2ddc7/src/python/bot/fuzzers/libFuzzer/engine.py#L283

Reply:
Done!